crypto/asn1/tasn_enc.c

```diff
@@ -79,20 +80,20 @@ static int asn1_item_flags_i2d(ASN1_VALUE *val, unsigned char **out,
  * performs the normal item handling: it can be used in external types.
  */
 
-int ASN1_item_ex_i2d(ASN1_VALUE **pval, unsigned char **out,
+int ASN1_item_ex_i2d(const ASN1_VALUE **pval, unsigned char **out,
                      const ASN1_ITEM *it, int tag, int aclass)
 {
     const ASN1_TEMPLATE *tt = NULL;
     int i, seqcontlen, seqlen, ndef = 1;
     const ASN1_EXTERN_FUNCS *ef;
     const ASN1_AUX *aux = it->funcs;
-    ASN1_aux_cb *asn1_cb = 0;
+    ASN1_aux_const_cb *asn1_cb = 0;
 
     if ((it->itype != ASN1_ITYPE_PRIMITIVE) && !*pval)
         return 0;
 
     if (aux && aux->asn1_cb)
-        asn1_cb = aux->asn1_cb;
+        asn1_cb = aux->asn1_const_cb;
```

Comment:
This represents an API break since anyone with old code that only defines asn1_cb (and not asn1_const_cb) will suddenly stop working - won't it??

Reply:
Please let's continue discussing this in the above sub-thread started on Jan 20 because there we have already some relevant context, in particular an important question on this I stated on Jan 23, which so far unanswered. I'll copy your new comment to there and handle it there.

Reply:
@mattcaswell, I've meanwhile added a backward compatibility flag to solve this issue as suggested by @levitte, see https://github.com/openssl/openssl/pull/8029#discussion_r255817443